src/components/ControlledTextInputField/ControlledTextInputField.js

```diff
@@ -33,15 +39,23 @@ const ControlledTextInputField = ({ control, name, label, placeholder }) => {
       <Controller
         control={control}
         name={name}
+        rules={{
+          ...rules,
+          ...(required && { required: "This field is required" }),
+        }}
```

Comment:
Swap the order of this so the parent component can override the rule if they want